### PR TITLE
Update Model Objects

### PR DESCRIPTION
**Pull request by HaroldFS**
- product
- order
- addressDetail
- cardDetail
- userProfile

### CODE REVIEW

File: swift/ios-shoppe-demo/Model/Product.swift [hunk 1]
```diff
@@ -21,8 +21,12 @@ class Product {
     init(_ dict: [String: Any]) {
         self.title = dict["title"] as? String ?? ""
         self.description = dict["description"] as? String ?? ""
-        self.price = dict["price"] as? Double ?? -1.0
+        self.price = Decimal(floatLiteral: dict["price"] as? Double ?? -1.0)
```
Comment (van-fs) [inline swift/ios-shoppe-demo/Model/Product.swift:24]: I noticed else where the Doubles are now Decimals.  But this line still looks like it casts `as?` a `Double`.  Is that intentional?

**Review by van-fs — APPROVED** [review]
LGTM; see my one comment about the Double.

File: swift/ios-shoppe-demo/Model/Product.swift (hunk 1 above)
Comment (HaroldFS) [reply swift/ios-shoppe-demo/Model/Product.swift:24]: no I will fix that thanks so much for catching it!

Comment (HaroldFS) [reply swift/ios-shoppe-demo/Model/Product.swift:24]: Oh actually I ran into an issue and it was not recognized as a decimal so I needed to cast as a Double the initialize as a Decimal.

**Pull request merged**